src/screens/Voucher/VoucherIndexScreen.tsx

```diff
@@ -0,0 +1,99 @@
+import { StackScreenProps } from '@react-navigation/stack';
+import _uniqBy from 'lodash/uniqBy';
+import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { useQuery } from 'react-apollo';
+import { RefreshControl } from 'react-native';
+
+import { NetworkContext } from '../../NetworkProvider';
+import {
+  EmptyMessage,
+  ListComponent,
+  LoadingSpinner,
+  RegularText,
+  Wrapper
+} from '../../components';
+import { colors, texts } from '../../config';
+import { graphqlFetchPolicy, parseListItemsFromQuery } from '../../helpers';
+import { QUERY_TYPES, getFetchMoreQuery, getQuery } from '../../queries';
+
+export const VoucherIndexScreen = ({ navigation, route }: StackScreenProps<any>) => {
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
+
+  const [refreshing, setRefreshing] = useState(false);
+
+  const query = route.params?.query ?? '';
+
+  const { data, loading, fetchMore, refetch } = useQuery(getQuery(query), { fetchPolicy });
+
+  const listItems = useMemo(() => {
+    return parseListItemsFromQuery(QUERY_TYPES.VOUCHERS, data, undefined, {
+      withDate: false
+    });
+  }, [data, query]);
```

Comment:
there is something missing to filter the latest x vouchers instead of all, right?
do we need that from the server as query? to be able to limit on 20 and sort by creation date?

Reply:
Yes, you are right. Here I added limit and order to queryVariables. It works fine. But the query still has different problems. 

https://github.com/smart-village-solutions/smart-village-app-app/pull/988/commits/b1ca1b39b81f0f495e65c20261bd6cedb7b4be0b